### ISSUE TITLE
Question about the R2R dataset

### ISSUE THREAD

**Issue by dadadadawd** [CrystalSixone/VLN-GOAT#18]
Thank you very much for your excellent work! 

I have a few questions I’d like to ask:

When I tried to reproduce your results on the R2R dataset, I got the following performance:

On the validation seen split, the best score was 81.49, while your paper reports 83.74.

On the validation unseen split, I got 77.74, which is very close to your reported 77.84.
However, the SPL scores were 2–3 points lower than those in the paper.
I used exactly the same parameters as provided in the code, and the experiments were run on an A800 GPU.
I’d like to ask:

Could this gap be due to hardware differences?

Or would the result be different if I run the experiment again?

Is some variation in the results expected?

I also noticed that the latter part of your architecture seems similar to DUET. However, you placed self-attention before cross-attention, while DUET first applies cross-attention and then self-attention.

Is there a specific reason for this order?

What impact does it have on the model’s behavior?

Thanks again for your time and for your excellent work!

**Comment by CrystalSixone** [comment]
Hi! Thanks so much for your interest in our work! \:D

Regarding your questions:

1. Performance gap: Yes, performance can vary slightly due to hardware differences or random seeds. We've observed a standard deviation of about 1–2% across GPUs. On the same machine with identical hyperparameters and seed, results should be consistent. Also, since `log_every` is set to 1000, you might try reducing the logging frequency near the peak iteration to avoid missing better checkpoints.

2. Self-attention before cross-attention: Great observation! Our architecture follows RoBERTa's design (see [this line](https://github.com/huggingface/transformers/blob/main/src/transformers/models/roberta/modeling_roberta.py#L507)), where self-attention comes before cross-attention. DUET does the reverse, but we didn’t run ablations on the order. We kept RoBERTa’s structure to stay aligned with its pretraining behavior, though exploring the impact of order is an interesting direction.

Hope this helps, and thank you again for the thoughtful feedback!

**Comment by dadadadawd** [comment]
Thank you for your response. I have one more question. When I was running the REVERIE dataset, the fine-tuning script included the line:
aug_file=${DATA_ROOT}/REVERIE/annotations/REVERIE_aug_train_enc.json
However, this file doesn't exist. I replaced it with:
aug_file=${DATA_ROOT}/REVERIE/annotations/REVERIE_train_aug_roberta_enc.json
Is this the correct replacement? Thanks again for your help!

**Comment by zhaoxingZ** [comment]
I also encountered the same issue and used the same solution - I'd like to know if this is the correct approach. Thanks a lot!
`No such file or directory: '../datasets/REVERIE/annotations/REVERIE_aug_train_enc.json'`

**Comment by CrystalSixone** [comment]
@dadadadawd @zhaoxingZ 
Yes, please modify the path to the released one. Thank you for your reminder.
